library/bind.c

```diff
@@ -381,7 +381,7 @@ static void process_dial(struct binding_s *b, message *msg) {
     bool caller_id_sent = message_get_bytes_header(msg, CallerIdHeader, &source_identity, &source_identity_sz);
 
     ziti_client_ctx clt_ctx = {0};
-    message_get_bytes_header(msg, AppDataHeader, (uint8_t **) &clt_ctx.app_data, &clt_ctx.app_data_sz);
+    message_get_bytes_header(msg, AppDataHeader, &clt_ctx.app_data, &clt_ctx.app_data_sz);
```

Comment:
it would be safer to make a copy here -- otherwise app_data points to garbage once msg is discarded

Reply:
Just to be clear, the variable scoping has not been changed in this PR. 

I believe `cit_ctx->app_data` will be assigned the pointer from the message header. `clt_ctx` is a stack variable, which is only passed to the client callback. We do [document that the client context is only valid while the callback is on the stack](https://github.com/openziti/ziti-sdk-c/blob/main/includes/ziti/ziti.h#L337).

Reply:
I think to make it more clear both `clt_ctx` and `ctl_ctx.app_data` should be `const` in the callback